src/features/security-cloud-game/SecurityCloudGame.js

```diff
@@ -1,82 +1,115 @@
-        cloudSize: PropTypes.number,
-        /** Height to set the game to */
-        height: PropTypes.number.isRequired,
-        /** Function to call when the `DragCloud` is successfuly dropped onto the `DropCloud` */
-        onValidDrop: PropTypes.func,
-        /** Width to set the game to */
-        width: PropTypes.number.isRequired,
-    };
+// pick these numbers to balance accessibility and game complexity
+const CLOUD_SIZE_RATIO = 5;
+const GRID_TRACK_SIZE_RATIO = 20;
+
+const SecurityCloudGame = ({ height, intl, onValidDrop, width }) => {
+    const [cloudSize, setCloudSize] = useState(null);
+    const [gridTrackSize, setGridTrackSize] = useState(null);
+    const [dropCloudPosition, setDropCloudPosition] = useState(null);
+    const [dragCloudPosition, setDragCloudPosition] = useState(null);
+    const [gameBoardHeight, setGameBoardHeight] = useState(null);
+    const [gameState, setGameState] = useState({
+        isValidDrop: false,
+        isOverlap: false,
+        liveText: '',
+    });
+
+    const messageElementRef = useRef();
+    // to handle resize events
+    const prevHeight = useRef();
+    const prevWidth = useRef();
+
+    useLayoutEffect(() => {
+        const { current: messageElement } = messageElementRef;
+        const newGameBoardHeight = height - messageElement.getBoundingClientRect().height;
+        const smallerDimension = Math.min(newGameBoardHeight, width);
+        const newGridTrackSize = smallerDimension / GRID_TRACK_SIZE_RATIO;
+        const newCloudSize = smallerDimension / CLOUD_SIZE_RATIO;
+        setGameBoardHeight(newGameBoardHeight);
+        setGridTrackSize(newGridTrackSize);
+        setCloudSize(newCloudSize);
+
+        messageElement.setAttribute(
+            'aria-label',
+            intl.formatMessage(
+                messages.accessibilityInstructions,
+                getGridPosition({ x: width - newCloudSize, y: newGameBoardHeight - newCloudSize }, newGridTrackSize),
+            ),
+        );
+        messageElement.focus();
+    }, [height, intl, width]);
 
-    static defaultProps = {
-        cloudSize: 64,
-    };
+    useEffect(() => {
+        if (!gameBoardHeight) {
+            return;
+        }
+
+        const heightRatio = prevHeight.current ? gameBoardHeight / prevHeight.current : 1;
+        const widthRatio = prevWidth.current ? width / prevWidth.current : 1;
 
-    constructor(props) {
-        super(props);
-        this.occupiedRegions = [];
-        this.state = {
-            isValidDrop: false,
-            isOverlap: false,
-        };
-
-        this.onDrag = throttle(this.onDrag, 100, {
-            leading: true,
-            trailing: true,
+        let newDropCloudPosition;
+        setDropCloudPosition(prevPos => {
+            newDropCloudPosition = !prevPos
+                ? getRandomCloudPosition(cloudSize, gameBoardHeight, width)
+                : { x: prevPos.x * widthRatio, y: prevPos.y * heightRatio };
```

Comment:
Yes, it's ok if x and y are not integers.
It's possible but it doesn't really matter. Since `isOverlap` can only change when `updatePosition()` is called, the user would still have to move the DragCloud again anyway.